components/brave_wallet/browser/brave_wallet_provider_events_observer.h

```diff
@@ -0,0 +1,18 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_BRAVE_WALLET_BROWSER_BRAVE_WALLET_PROVIDER_EVENTS_OBSERVER_H_
+#define BRAVE_COMPONENTS_BRAVE_WALLET_BROWSER_BRAVE_WALLET_PROVIDER_EVENTS_OBSERVER_H_
+
+#include <string>
+
+namespace brave_wallet {
+class BraveWalletProviderEventsObserver {
+ public:
+  virtual void ChainChangedEvent(const std::string& chain_id) = 0;
```

Comment:
It seems like this is defeating the purpose of having events because now we still need a new method for every event on several different classes and interfaces. Wouldn't it be simpler to just have `OnEvent` with the event type and associated data just like you have in `FireEvent`?

Reply:
it's like having an observer with a one single method `OnSomething()`. Not all events exposed to browser process.

Reply:
plus they have diff arguments to pass